contracts/token/ERC1155/IERC1155.sol

```diff
@@ -1,17 +1,18 @@
 // SPDX-License-Identifier: MIT
 // OpenZeppelin Contracts (last updated v4.7.0) (token/ERC1155/IERC1155.sol)
 
-pragma solidity ^0.8.0;
+pragma solidity ^0.8.18;
 
 import "../../utils/introspection/IERC165.sol";
+import "../../interfaces/draft-IERC6093.sol";
 
 /**
  * @dev Required interface of an ERC1155 compliant contract, as defined in the
  * https://eips.ethereum.org/EIPS/eip-1155[EIP].
  *
  * _Available since v3.1._
  */
-interface IERC1155 is IERC165 {
+interface IERC1155 is IERC165, ERC1155Errors {
```

Comment:
Is `ERC1155Errors` an interface ? If so, it should be name `IERC1155Errors`

Reply:
Also, `IERC1155` should remain as close as possible to the definition provided by EIP-1155. Anything on top (such as using the EIP-6093 errors) is an implementation choice that should be part of `ERC1155`

Reply:
> Also, IERC1155 should remain as close as possible to the definition provided by EIP-1155.

Is there any reason for this? I don't see why this is imperative.

> Anything on top (such as using the EIP-6093 errors) is an implementation choice that should be part of ERC1155

I agree developers might like to keep the implementation compliant with the EIP, but this is not forcing IERC1155 implementers to use the `ERC1155Errors`, they can just ignore them and it doesn't increment bytecode size as far as I know.

Reply:
Some project will use our interfaces as "references". For example in the subgraphs we use `IERC1155` as a model of what ERC1155 tokens will expose. If there is anything in the interface that is not in the standard, it can create confusion.

Here you have a good example with IERC165. Its here because 1155 requires it. The consequence is that anyone that calls a 1155 knows that they can call the "supportsInterface" function. It must be here.

While custom errors are not interfaces that one would call, if they are added to the interface they end up in the ABI and someone might expect all contracts that follow IERC1155 to use them. That is not the case.

I very strongly believe that IERC20 / IERC721 / IERC1155 must not be modified at all.